client/components/mma/cancel/CancellationReasonSelection.tsx

```diff
@@ -294,47 +291,95 @@ const ReasonPicker = (props: ReasonPickerProps) => {
 	);
 };
 
-const ReasonPickerRenderer =
-	(
-		productType: ProductTypeWithCancellationFlow,
-		productDetail: ProductDetail,
-	) =>
-	(apiResponse: CancellationDateResponse) => {
+interface ReasonPickerWithCancellationDateProps {
+	productType: ProductTypeWithCancellationFlow;
+	productDetail: ProductDetail;
+}
+
+const ReasonPickerWithCancellationDate = ({
+	productType,
+	productDetail,
+}: ReasonPickerWithCancellationDateProps) => {
+	const {
+		data: cancellationDateResponse,
+		loadingState,
+	}: {
+		data: CancellationDateResponse | null;
+		loadingState: LoadingState;
+	} = useAsyncLoader(
+		cancellationDateFetcher(productDetail.subscription.subscriptionId),
+		JsonResponseHandler,
+	);
+
+	if (loadingState == LoadingState.HasError) {
+		return <GenericErrorScreen />;
+	}
+
+	if (loadingState == LoadingState.IsLoading) {
 		return (
-			<ReasonPicker
-				productType={productType}
-				productDetail={productDetail}
-				chargedThroughCancellationDate={
-					apiResponse.cancellationEffectiveDate
-				}
+			<DefaultLoadingView
+				loadingMessage={`Checking your ${
+					productType.shortFriendlyName || productType.friendlyName
+				} details...`}
 			/>
 		);
-	};
+	}
+
+	if (cancellationDateResponse === null) {
+		return <GenericErrorScreen />;
+	}
+
+	// offer choice if not trial period or lead time, and startPageOfferEffectiveDateOptions config set to true
+	const shouldOfferEffectiveDateOptions =
+		!isNaN(
+			Date.parse(cancellationDateResponse.cancellationEffectiveDate),
+		) && productType.cancellation.startPageOfferEffectiveDateOptions;
```

Comment:
You don't really need this check now as this component is only rendered if `startPageOfferEffectiveDateOptions` is true.

Reply:
amended